fairscale/experimental/wgit/repo.py

```diff
@@ -0,0 +1,131 @@
+
+        if not self._exists() and init:
+            # No weigit repo exists and is being initialized with init=True
+            # Make .wgit directory, create sha1_refs and metadata file
+            self._wgit_dir.mkdir(exist_ok=True)
+            self._metadata_file.touch(exist_ok=False)
+            self._sha1_ref.touch(exist_ok=False)
+
+            # # Make the .wgit a git repo
+            gitignore_files = [self._sha1_store_path.name, self._sha1_ref.name]
+            self._pygit = PyGit(self.wgit_parent.joinpath(self._wgit_dir), gitignore=gitignore_files)
+
+            # Initializing sha1_store only after wgit has been initialized!
+            self._sha1_store = SHA1_store(self._wgit_dir, self._metadata_file, self._sha1_ref, init=True)
+        elif self._exists() and init:
+            # if weigit repo already exists and init is being called, wrap the existing .wgit/.git repo with PyGit
+            self._sha1_store = SHA1_store(
+                self._wgit_dir,
+                self._metadata_file,
+                self._sha1_ref,
+            )
+            self._pygit = PyGit(self.wgit_parent.joinpath(self._wgit_dir))
+
+        elif self._exists() and not init:
+            # weigit exists and non-init commands are triggered
+            self._sha1_store = SHA1_store(
+                self._wgit_dir,
+                self._metadata_file,
+                self._sha1_ref,
+            )
+            self._pygit = PyGit(self.wgit_parent.joinpath(self._wgit_dir))
+
+        else:
+            # weigit doesn't exist and is not trying to be initialized (triggers during non-init commands)
+            sys.stderr.write("fatal: not a wgit repository!\n")
+
+    def add(self, file_path: str) -> None:
+        """Adds a file to the wgit repo"""
+        if self._exists():
+            self._sha1_store.add(file_path)  # add the filefile to the sha1_store
+            self._pygit.add()  # add to the .wgit/.git repo
+
+    def commit(self, message: str) -> None:
+        """Commits staged changes to the repo"""
+        if self._exists():
+            self._pygit.commit(message)
+
+    def status(self) -> None:
+        """Skeleton"""
+        if self._exists():
+            print("wgit status")
+
+    def log(self, file: str) -> None:
+        """Returns the WeiGit log of commit history."""
+        if self._exists():
+            if file:
+                print(f"wgit log of the file: {file}")
+            else:
+                print("wgit log")
+
+    def checkout(self, sha1: str) -> None:
+        """Checkout a previously commited version of the checkpoint"""
+        if self._exists():
+            print("wgit checkout: sha1")
+
+    def compression(self) -> None:
+        """Not Implemented: Compression functionalities"""
+        print("Not Implemented!")
+
+    def checkout_by_steps(self) -> None:
+        """Not Implemented: Checkout by steps"""
+        print("Not Implemented!")
```

Comment:
you could `raise NotImplementedError()`. I remember this is a standard python exception for it.